From e56f6aa08c25276398f6170ded46653e7301b8c0 Mon Sep 17 00:00:00 2001
From: Sam Leffler <sleffler@chromium.org>
Date: Tue, 19 Jul 2011 15:27:11 -0700
Subject: [PATCH] CHROMIUMOS: wpa_supplicant: add support for tx aborting a
 scan

Add support to mark a scan request so that outbound traffic will abort the
operation and report whether a completed scan was aborted.  This mechanism
is implemented for nl80211 by setting NL80211_SCAN_FLAG_TX_ABORT in a
new NL80211_ATTR_SCAN_FLAGS property.

BUG=chromium-os:11485
TEST=manual:depends on other changes

Change-Id: I348943800428dc8db1ca0c07996421cb20a7f5e3
Reviewed-on: http://gerrit.chromium.org/gerrit/4346
Reviewed-by: Gary Morain <gmorain@chromium.org>
Reviewed-by: Paul Stewart <pstew@chromium.org>
Tested-by: Sam Leffler <sleffler@chromium.org>
Commit-Ready: Sam Leffler <sleffler@chromium.org>
---
 src/drivers/driver.h  | 2 ++
 wpa_supplicant/scan.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/drivers/driver.h b/src/drivers/driver.h
index 6af7294..f413bf1 100644
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -241,11 +241,13 @@ struct wpa_scan_res {
  * @res: Array of pointers to allocated variable length scan result entries
  * @num: Number of entries in the scan result array
  * @fetch_time: Time when the results were fetched from the driver
+ * @aborted: Whether the scan was aborted
  */
 struct wpa_scan_results {
 	struct wpa_scan_res **res;
 	size_t num;
 	struct os_reltime fetch_time;
+	int aborted;
 };
 
 /**
diff --git a/wpa_supplicant/scan.c b/wpa_supplicant/scan.c
index debceb9..d715b69 100644
--- a/wpa_supplicant/scan.c
+++ b/wpa_supplicant/scan.c
@@ -1796,6 +1796,8 @@ wpa_supplicant_get_scan_results(struct wpa_supplicant *wpa_s,
 	      compar);
 	dump_scan_res(scan_res);
 
+	scan_res->aborted = (info && info->aborted);
+
 	wpa_bss_update_start(wpa_s);
 	for (i = 0; i < scan_res->num; i++)
 		wpa_bss_update_scan_res(wpa_s, scan_res->res[i],
-- 
2.2.0.rc0.207.ga3a616c

